app/reducers/threads.ts

```diff
@@ -0,0 +1,56 @@
+import { createSelector } from '@reduxjs/toolkit';
+import { Thread } from 'app/actions/ActionTypes';
+import createEntityReducer from 'app/utils/createEntityReducer';
+import { mutateComments, selectCommentEntities } from './comments';
+import type { ID } from 'app/models';
+
+const mutate = mutateComments('threads');
+
+export default createEntityReducer({
+  key: 'threads',
+  types: {
+    fetch: Thread.FETCH_ALL,
+    mutate: Thread.CREATE,
+    delete: Thread.DELETE,
+  },
+  mutate,
+});
+
+export const selectThreadsByForumId = (forumId: ID) =>
+  createSelector(
+    (state) => state.threads.byId,
+    (state) => state.threads.items,
+    (threadsById, threadIds) => {
+      const filteredThreadIds = threadIds.filter((id) => {
+        const thread = threadsById[id];
+        return thread && thread.forum === forumId;
+      });
+
+      return filteredThreadIds.map((id) => threadsById[id]);
```

Comment:
I'm just writing a suggestion out of the top of my head, so idk if it runs - but this is where I thought it would be beneficial to have the `/forums/<id>/threads` API structure

```suggestion
    (state) => state.forums.byId,
  (state, props) => props.forumId,
  (forumsById, forumId) => forumsById[forumId].threads
```